modules/reporting/lib/open_project/reporting/cost_entry_xls_table.rb

```diff
@@ -131,14 +131,16 @@ def cost_entry_attributes
     %i[user_id activity_id entity_gid comments project_id]
   end
 
-  # Returns the results of the query sorted by date the time was spent on and by id
+  # Returns the results of the query sorted by date the time was spent on and name
   def sorted_results
     query
       .each_direct_result
       .map(&:itself)
       .group_by { |r| r.fields["spent_on"] }
       .sort
-      .flat_map { |_, date_results| date_results.sort_by { |r| r.fields["id"] } }
+      .flat_map do |_, date_results|
+        date_results.sort_by { |r| User.find_by(id: r.fields["user_id"])&.name&.downcase || "" }
```

Comment:
Calling `User.find_by` inside a sort will generate n+1 queries. I'm not sure what a typical recordset looks like, but this might eventually become a performance concern.

To preload users, you would first need to extract the IDs:


```ruby
results = query.each_direct_result.to_a
user_ids = results.filter_map { |r| r.fields["user_id"] }.uniq
users_by_id = User.where(id: user_ids).pluck(:id, :name).to_h
```

then

```suggestion
        users_by_id[r.fields["user_id"]] &.downcase || ""
```

Reply:
Thanks! I couldn't use your code exactly because `name` is a model method, but I believe what I pushed now should be fine too. Could you take another look please?

Reply:
Yes, I missed that in my suggestion. Looks good now.